src/lib/MapTypeButton/MapTypeButton.svelte

```diff
@@ -0,0 +1,53 @@
+<script>
+    import { createEventDispatcher } from 'svelte';
+
+    const dispatch = createEventDispatcher();
+    export let mapTypeId = 'roadmap';
+
+    const mapTypes = ['roadmap', 'satellite', 'hybrid', 'terrain'];
+
+    function cycleMapType() {
+        const currentIndex = mapTypes.indexOf(mapTypeId);
+        const nextIndex = (currentIndex + 1) % mapTypes.length;
+        mapTypeId = mapTypes[nextIndex];
+        dispatch('mapTypeChanged', mapTypeId);
+    }
+
+    function getMapTypeIcon(type) {
+        switch (type) {
+            case 'roadmap': return '🗺️';
+            case 'satellite': return '🛰️';
+            case 'hybrid': return '🌎';
+            case 'terrain': return '⛰️';
+            default: return '🗺️';
+        }
+    }
+</script>
+
+<button class="map-type-button" on:click={cycleMapType}>
+    {getMapTypeIcon(mapTypeId)}
+</button>
+
+<style>
+	.map-type-button {
```

Comment:
To maintain consistency, you can use the Tailwind CSS inline at the button.

Reply:
Okay, no problem. We will add the button CSS code into a global CSS file in the future.